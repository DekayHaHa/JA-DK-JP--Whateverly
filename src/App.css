html { height: 100%; }
body { height: 100%; background: black; }
#root { height: 100%; }

.app {
  position: fixed;
  width: 100%;
  height: 100%;
  /*min-height: 100vh;*/
  /*background: hsl(240, 25%, 35%);*/
  background: url("./imgs/helmet.gif");
  background-size: cover;
}

input,
button {
  font-family: 'Audiowide', cursive;
}