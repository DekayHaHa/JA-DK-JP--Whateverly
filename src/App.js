import React, { Component } from 'react';
import Header from './Header.js';
import GameShelf from './GameShelf.js';
import Footer from './Footer.js';
import './App.css';

class App extends Component {
  constructor () {
    super();
    this.state = { 
      games: [],
      mainGenres: [],
      genres: [],
      platforms: [],
      genresToFilter: [],
      platformsToFilter: [],
      multiplayer: [null, true, false]
     }
  }
  componentDidMount = () => {
    fetch('http://whateverly-datasets.herokuapp.com/api/v1/games1811')
      .then(response => response.json())
      .then(games => {
        this.setState({
          games: games.games1811
        })
      })
      .then(() => this.getFilters())
      .catch(error => {
        throw new Error(error)
      })
    fetch('http://whateverly-datasets.herokuapp.com/api/v1/genres1811')
      .then(response => response.json())
      .then(genres => {
        this.setState({
          mainGenres: genres.genres1811
        })
      })
      .then(() => this.getFilters())
      .catch(error => {
        throw new Error(error)
      })
  }
  getFilters = () => {
    let filters = [];
    let filtersTwo = [];
    this.state.games.forEach(game => {
      game.platforms.forEach(val => {
        console.log("in filter1")
        if (!filters.includes(val)){
          filters.push(val)
        }
      })
    })
    this.state.games.forEach(game => {
      game.genres.forEach(val => {
        if (!filtersTwo.includes(val)){
          filtersTwo.push(val)
        }
      })
    })
    this.setState({platforms: filters})
    this.setState({genres: filtersTwo})
                  
  }
  updateState = (newState) => {
    this.setState(newState);
  }
 
  render() {
    console.log("games", this.state.games);
    console.log("plat", this.state.platforms);
    console.log("genre", this.state.genres);
    return (
      <div className="app">
        <Header />
        <GameShelf 
          games={this.state.games}/>
        <Footer {...this.state}
                updateState={this.updateState}/>
      </div>
    );
  }
}

export default App;