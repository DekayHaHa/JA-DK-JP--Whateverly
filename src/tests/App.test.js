import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it ('should have default state', () => {
    expect(wrapper.state()).toEqual({ 
      games: [],
      mainGenres: [],
      genres: [],
      platforms: [],
      genresToFilter: [],
      platformsToFilter: [],
      filteredGames: [],
      multiplayer: [null, true, false],
      titleFilter: '',
      genresToFilter: []
     });
  })
})
