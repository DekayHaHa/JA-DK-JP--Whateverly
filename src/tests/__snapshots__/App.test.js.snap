// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshop 1`] = `
<div
  className="app"
>
  <Header />
  <GameShelf
    filteredGames={Array []}
    games={Array []}
    genres={Array []}
    genresToFilter={Array []}
    mainGenres={Array []}
    multiplayer={
      Array [
        null,
        true,
        false,
      ]
    }
    platforms={Array []}
    platformsToFilter={Array []}
    titleFilter=""
  />
  <Footer
    filteredGames={Array []}
    games={Array []}
    genres={Array []}
    genresToFilter={Array []}
    mainGenres={Array []}
    multiplayer={
      Array [
        null,
        true,
        false,
      ]
    }
    platforms={Array []}
    platformsToFilter={Array []}
    titleFilter=""
    updateState={[Function]}
  />
</div>
`;
